src/waveform/renderers/qopengl/waveformrendererrgb.cpp

```diff
@@ -0,0 +1,297 @@
+#include "waveform/renderers/qopengl/waveformrendererrgb.h"
+
+#include "fixedpointcalc.h"
+#include "track/track.h"
+#include "util/math.h"
+#include "waveform/waveform.h"
+#include "waveform/waveformwidgetfactory.h"
+#include "waveform/widgets/qopengl/waveformwidget.h"
+#include "widget/wskincolor.h"
+#include "widget/wwidget.h"
+
+using namespace qopengl;
+
+WaveformRendererRGB::WaveformRendererRGB(
+        WaveformWidgetRenderer* waveformWidget)
+        : WaveformRendererSignalBase(waveformWidget) {
+}
+
+WaveformRendererRGB::~WaveformRendererRGB() {
+}
+
+void WaveformRendererRGB::onSetup(const QDomNode& node) {
+    Q_UNUSED(node);
+}
+
+void WaveformRendererRGB::initializeGL() {
+    QString vertexShaderCode =
+            "\
+uniform mat4 matrix;\n\
+attribute vec4 position;\n\
+attribute vec3 color;\n\
+varying vec3 vcolor;\n\
+void main()\n\
+{\n\
+    vcolor = color;\n\
+    gl_Position = matrix * position;\n\
+}\n";
+
+    QString fragmentShaderCode =
+            "\
+varying vec3 vcolor;\n\
+void main()\n\
+{\n\
+    gl_FragColor = vec4(vcolor,1.0);\n\
+}\n";
+
+    if (!m_shaderProgram.addShaderFromSourceCode(QOpenGLShader::Vertex, vertexShaderCode)) {
+        return;
+    }
```

Comment:
these should probably all be `VERIFY_OR_DEBUG_ASSERT`s.

Reply:
agreed